### DISCUSSION TITLE
Please use standardized `_FILE` method for Docker secrets.

### DISCUSSION THREAD

**Discussion by andrewvaughan** [mealie-recipes/mealie#3773] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

The current [documented recommendation](https://docs.mealie.io/documentation/getting-started/installation/backend-config/#docker-secrets) to make use of Docker secrets doesn't follow [Docker Compose best practices](https://docs.docker.com/compose/use-secrets/).

### Please provide a concise description of the feature that would resolve your issue.

Specifically, Docker recommends that any secrets environment variable be supplemented with a `_FILE` version of that variable and the application should first-look to see if that file is set, loading the secret from the provided filename, and only defaulting to the original variable if no secret is set.

The Mealie Docker configuration should be modified to include the following, additional environment variables dedicated for secrets, similar to how [WordPress](https://github.com/docker-library/wordpress/blob/8c2ded17022e67c6accacfae637c5a577ce169af/wp-config-docker.php#L26-L40) and [MySQL](https://github.com/docker-library/mysql/blob/1a703318803fa85bf69cb5d2e5b3f1c4087627b5/docker-entrypoint.sh#L24-L43) implement this standard (both of these links show two different styles of implementation of this functionality):

- `POSTGRES_USER_FILE`
- `POSTGRES_PASSWORD_FILE`
- `POSTGRES_SERVER_FILE`
- `POSTGRES_PORT_FILE`
- `POSTGRES_DB_FILE`

This should be modified so that any environment variable above, if set, will load the password from a file in the environment, only deferring to the original environment variable if this doesn't exist. This is also backwards compatible to all current configurations.

### Please consider and list out some caveats or tradeoffs made in your design decision

This is actually quite critical, because for any individuals using a Postgres database for multiple clients in a Docker Compose file, the requirement that Mealie has to dominate the `POSTGRES_PASSWORD` secret name can be a problem for multi-tenant compose files. Any other container that uses `POSTGRES_PASSWORD` will now have to share that secret with Mealie, which should have apparent security risks.

Please note that I have only considered the `POSTGRES` integration in this request, but this likely applies to any area that users may wish to better-use secrets in their Mealie implementation, such as for SMTP configurations, LDAP configurations/certificates, OIDC configurations, and OpenAI integrations. I have limited the scope of this particular discussion to Postgres, as that is the only secret referenced in the documentation (and the only one currently within my scope of need).

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Already discussed a bit in https://github.com/mealie-recipes/mealie/issues/3606, but this makes sense to me. If you plan on working on this (or anyone), just be sure not to break existing installs.

> this likely applies to any area that users may wish to better-use secrets in their Mealie implementation

Here is a list of secrets used by Mealie:
- POSTGRES_PASSWORD
- SMTP_PASSWORD
- LDAP_QUERY_PASSWORD
- OPENAI_API_KEY
